cloverage/src/cloverage/coverage.clj

```diff
@@ -134,19 +134,25 @@
       namespaces)))
 
 (defn load-namespaces [{:keys [extra-test-ns ns-regex test-ns-regex ns-exclude-regex src-ns-path test-ns-path]} add-nses]
-  (let [include      (-> src-ns-path
-                         (find-nses ns-regex)
-                         (remove-nses ns-exclude-regex))
-        namespaces   (concat add-nses include)
-        test-nses    (concat extra-test-ns (find-nses test-ns-path test-ns-regex))
-        ordered-nses (dep/in-dependency-order (map symbol namespaces))]
+  ;; Find all namespaces in the classpath (and any additional ones added by `add-nses`) and then sort them in order so
+  ;; dependencies get loaded first.
+  (let [all-nses    (dep/in-dependency-order (map symbol (distinct (concat add-nses (find-nses src-ns-path nil)))))
+        ;; calculate the set of namespaces we're planning on instrumenting
+        include     (-> src-ns-path
+                        (find-nses ns-regex)
+                        (remove-nses ns-exclude-regex))
+        instrument? (set (map symbol (concat add-nses include)))
+        ;; Now get a sequence of namespaces to instrument in the dependency order. Do this by taking the entire sorted
+        ;; sequence of `all-nses` and filtering out ones we don't want to instrument
+        namespaces  (filter instrument? all-nses)
+        test-nses   (concat extra-test-ns (find-nses test-ns-path test-ns-regex))]
```

Comment:
One issue I ran into was that `in-dependency-order` only considered the namespaces being instrumented; if you were instrumenting 2 out of 10 and there were some relevant dependencies in the other 8 the resultant order would be incorrect. This change first determines the correct order for all namespaces (e.g. all 10 out of 10) and then determines the correct order for the ones we want to instrument